x-pack/plugins/security_solution/server/endpoint/routes/actions/list_handler.ts

```diff
@@ -73,7 +80,21 @@ export const actionListHandler = (
         return res.notFound({ body: 'index_not_found_exception' });
       }
 
+      // verify feature flag for sentinel_one `aaentType`
+      const agentTypes = formatAgentTypeValues(_agentTypes);
+      if (
+        !endpointContext.experimentalFeatures.responseActionsSentinelOneV1Enabled &&
```

Comment:
That would also result in the custom API error.

Reply:
Thanks! Shouldn't we return at least 'endpoint' results? I think that is how the other filtering dropdowns work. What do you think?

Reply:
No the other filters behave the same way. Also, if we quietly processed only valid values ignoring the invalid values that would not inform the user that they are trying to do something that is restricted.

Reply:
@tomsonpl ,

It would not be a good thing to return "partial" results to the caller of the API if we did not support a certain type.  Example: user request `endpoint` and `sentinel_one` agent types, but we return only `endpoint`. What does that actually mean from the caller's standpoint? that there are no `sentinel_one` response actions for that agent type? or that we don't support it.

Just like our other schema validations, if a give value is not supported, we should error back. This would normally be handled by the schema itself, but given that out schemas as defined as `const`'s rather than us having a `getter` factory function that returns the schema, we can't really implement it there. so the check here and associated error is mimicking what the Schema validation would have done.

Hope that makes sense

Reply:
Thanks for explaining with example @paul-tavares. ❤️

Reply:
Hmm I am not sure if I agree with that, let me explain why: 
Eg. with the manual and automated filter in history log, when we select manual, we get manual. when we select automated we get automated. But if we don't select, or select both - we get all actions, or at least whatever we can get. I don't see a reason why user should get an error when two options are chosen. The same with any filters for actions - we can toggle multiple actions, right? Why this one should be any different?
However it's possible that I am misunderstanding something, please help me understand :P

Reply:
@tomsonpl  What you're saying is on the point. Although I think you're confusing with using valid values vs throwing errors on invalid values. 

We should not throw errors on query params that do accept "valid" multiple values. For example if the query param has `isolate` and `get-file` (or `['isolate','get-file]`) for `commands` filter that works fine. However we should throw and error if the query param is `['isolate', 'get-file', 'non-allowed-text' ]`.

So here in this PR we're restricting the usage of `sentinel_one` agent type as it is currently behind a feature flag and is akin to the `non-allowed-text` above. If you do have the feature flag it should use the vallue the search query. 

If you don't have the feature flag then it should throw an error if you try to filter with `['endpoint', 'sentinel_one']` or with `['sentinel_one']`. 

When feature flag is off, and the API quietly processes the request without using the `sentinel_one` value , then as @paul-tavares pointed out, the user won't know if the data returned (without any `sentinel_one` results) from the search result is because there is no response data for `sentinel_one` or that we didn't use the filter param in the first place.

I hope that clears up the confusion.

Reply:
Oh yes, thanks for clarification, I think we mixed up too concepts here:
1. fetching actions based on agent_types.
2. throwing error when the functionality is not enabled (feature flag set to false)

So just to clarify:
1. here, passing two agent_types will result in returning actions from both agents right? 
2. Is there any specific reason why we want to throw error on a not enabled functionality? Why would user even try to set sentinel_one? If there is a command/agent_type whatever different parameter that doesn't meet the schema - the action will throw the error, why do we want to specifically say that it's about sentinel_one here? 

Thanks!

Reply:
@tomsonpl 

1. Yes. Using multiple allowed values are used in the search query, and thus will return all results matching those values.
2. The schema is configured to allow `sentinel_one` and `endpoint` agent types and knows nothing about feature flag (_ff_).  Schema validation will throw error if one requested for any other values other than those two. The _ff_ is verified in the route handler, and since the _ff_ is specifically for `sentinel_one`,  that is why we want to report `sentinel_one` specific error if we tried to filter with `sentinel_one` and that _ff_ is off.

From re-reading  your question I sense that you are actually pointing to the fact that the error message should report allowed values instead when the _ff_ is disabled? So instead of `[request body.agentTypes]: sentinel_one is disabled'` it should be `[request body.agentTypes]: only endpoint agent is allowed'`? If that is the case then it is okay to change to that, for now. 

Although as the agent types list will grow and we won't be using _ffs_, it's better to report the non-allowed values in the error instead. Generally, I'd prefer we report the erroneous values in the error info.

Reply:
Alright now I get it. Not sure about the 2nd one still but it's alright as long as the 1st works with multiple values I am good with it. 
Thanks for the explanation :) 🚢 🇮🇹  :)